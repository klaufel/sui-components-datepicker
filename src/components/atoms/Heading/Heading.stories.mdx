import {Meta, Props, Story, Preview} from '@storybook/addon-docs/blocks'
import Heading from '.'

<Meta title="Atoms/Heading" component={Heading} />

# Heading

Let's define a story for our `Heading` component:

<Preview>
  <Story name="default">
      <Heading>Lorem ipsum dolor sit amet</Heading>
  </Story>
</Preview>

## Props

<Props of={Heading} />

## Sizes

Determines the sizes of the titles depending on the context or arrangement of the component.

#### Default size

<Preview>
  <Story name="size default">
    <Heading>Lorem ipsum dolor sit amet</Heading>
  </Story>
</Preview>

#### Small size

<Preview>
  <Story name="size small">
    <Heading size="small">Lorem ipsum dolor sit amet</Heading>
  </Story>
</Preview>

#### Huge size

<Preview>
  <Story name="size huge">
    <Heading size="huge">Lorem ipsum dolor sit amet</Heading>
  </Story>
</Preview>
