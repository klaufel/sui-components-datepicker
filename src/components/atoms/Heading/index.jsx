import React from 'react'
import PropTypes from 'prop-types'
import './index.scss'

const HEADINGS_TYPES = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'span']

const HEADINGS_SIZES = ['', 'small', 'huge']

const Heading = ({type, size, children, ...props}) => {
  const classname = `Heading${size && ` Heading--${size}`}`
  return React.createElement(type, {className: classname, ...props}, children)
}

Heading.propTypes = {
  type: PropTypes.oneOf(HEADINGS_TYPES),
  size: PropTypes.oneOf(HEADINGS_SIZES),
  children: PropTypes.node.isRequired
}

Heading.defaultProps = {
  type: 'h1',
  size: ''
}

export default Heading
