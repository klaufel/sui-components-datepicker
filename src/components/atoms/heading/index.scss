@import '~@schibstedspain/sui-theme/lib/settings-compat-v7/index';
@import '~@schibstedspain/sui-theme/lib/index';

$Heading-sizes: (
  'sm': 'h3',
  'md': 'h2',
  'lg': 'h1'
);

.Heading {
  @each $size, $value in $Heading-sizes {
    &--#{$size} {
      @if $value == 'h3' {
        @include h3;
      } @else if $value == 'h2' {
        @include h2;
      } @else if $value == 'h1' {
        @include h1;
      } @else {
        @error 'Unknown size #{size}.';
      }
    }
  }
}
