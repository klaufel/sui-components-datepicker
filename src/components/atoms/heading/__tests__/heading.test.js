import React from 'react'
import {render} from '@testing-library/react'
import Heading, {HEADINGS_TYPES, HEADINGS_SIZES} from '.'

const HeadingTypes = () =>
  HEADINGS_TYPES.map(type => (
    <Heading key={type} type={type}>
      Lorem ipsum dolor sit amet
    </Heading>
  ))

const HeadingSizes = () =>
  HEADINGS_SIZES.map(size => (
    <Heading key={size} size={size}>
      Lorem ipsum dolor sit amet
    </Heading>
  ))

describe('Heading', () => {
  test('it works default', () => {
    const {container} = render(<Heading>Lorem ipsum dolor sit amet</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with all sizes variations', () => {
    const {container} = render(<HeadingSizes />)
    expect(container).toMatchSnapshot()
  })

  test('it works with all types variations', () => {
    const {container} = render(<HeadingTypes />)
    expect(container).toMatchSnapshot()
  })
})
