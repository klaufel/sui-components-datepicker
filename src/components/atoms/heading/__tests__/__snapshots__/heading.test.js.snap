// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading it works default 1`] = `
<h1
  class="Heading Heading--md"
>
  Lorem ipsum dolor sit amet
</h1>
`;

exports[`Heading it works with all sizes variations 1`] = `
<div>
  <h1
    class="Heading Heading--sm"
  >
    Lorem ipsum dolor sit amet
  </h1>
  <h1
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </h1>
  <h1
    class="Heading Heading--lg"
  >
    Lorem ipsum dolor sit amet
  </h1>
</div>
`;

exports[`Heading it works with all types variations 1`] = `
<div>
  <h1
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </h1>
  <h2
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </h2>
  <h3
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </h3>
  <h4
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </h4>
  <h5
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </h5>
  <h6
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </h6>
  <span
    class="Heading Heading--md"
  >
    Lorem ipsum dolor sit amet
  </span>
</div>
`;
